### ISSUE TITLE
Fatal error: Call to a member function get_id() on array in class-wc-report-stock.php

### ISSUE THREAD

**Issue by foxvor** [woocommerce/woocommerce#18339]
After updating woocomerce to v3.2.5 I can't see my inventory report. I'm getting this php error:

Fatal error: Call to a member function get_id() on array in /wp-content/plugins/woocommerce/includes/admin/reports/class-wc-report-stock.php on line 79

Line 79 reads:

if ( ! $product || $product->get_id() !== $item->id ) {

the whole function is:
public function column_default( $item, $column_name ) {
		global $product;

		if ( ! $product || $product->get_id() !== $item->id ) {
			$product = wc_get_product( $item->id );
		}

		if ( ! $product ) {
			return;
		}

		switch ( $column_name ) {

			case 'product' :
				if ( $sku = $product->get_sku() ) {
					echo esc_html( $sku ) . ' - ';
				}

				echo esc_html( $product->get_name() );

				// Get variation data.
				if ( $product->is_type( 'variation' ) ) {
					echo '<div class="description">' . wp_kses_post( wc_get_formatted_variation( $product, true ) ) . '</div>';
				}
			break;

			case 'parent' :
				if ( $item->parent ) {
					echo esc_html( get_the_title( $item->parent ) );
				} else {
					echo '-';
				}
			break;

			case 'stock_status' :
				if ( $product->is_in_stock() ) {
					$stock_html = '<mark class="instock">' . __( 'In stock', 'woocommerce' ) . '</mark>';
				} else {
					$stock_html = '<mark class="outofstock">' . __( 'Out of stock', 'woocommerce' ) . '</mark>';
				}
				echo apply_filters( 'woocommerce_admin_stock_html', $stock_html, $product );
			break;

			case 'stock_level' :
				echo esc_html( $product->get_stock_quantity() );
			break;

			case 'wc_actions' :
				?><p>
					<?php
						$actions = array();
						$action_id = $product->is_type( 'variation' ) ? $item->parent : $item->id;

						$actions['edit'] = array(
							'url'       => admin_url( 'post.php?post=' . $action_id . '&action=edit' ),
							'name'      => __( 'Edit', 'woocommerce' ),
							'action'    => "edit",
						);

						if ( $product->is_visible() ) {
							$actions['view'] = array(
								'url'       => get_permalink( $action_id ),
								'name'      => __( 'View', 'woocommerce' ),
								'action'    => "view",
							);
						}

						$actions = apply_filters( 'woocommerce_admin_stock_report_product_actions', $actions, $product );

						foreach ( $actions as $action ) {
							printf(
								'<a class="button tips %1$s" href="%2$s" data-tip="%3$s">%4$s</a>',
								esc_attr( $action['action'] ),
								esc_url( $action['url'] ),
								sprintf( esc_attr__( '%s product', 'woocommerce' ), $action['name'] ),
								esc_html( $action['name'] )
							);
						}
					?>
				</p><?php
			break;
		}
	}

I've already updated to v3.2.6 and it's still the same. Anyone with this issue?

## WordPress Environment

woopress/woocommerce/myaccount/my-address.php
La  version 2.1.0 de woopress/woocommerce/single-product/add-to-cart/simple.php está obsoleta. La versión del núcleo es la 3.0.0
La  version 2.6.3 de woopress/woocommerce/single-product/product-image.php está obsoleta. La versión del núcleo es la 3.1.0
La  version 2.6.3 de woopress/woocommerce/single-product/product-thumbnails.php está obsoleta. La versión del núcleo es la 3.1.0
La  version 1.6.4 de woopress/woocommerce/single-product/related.php está obsoleta. La versión del núcleo es la 3.0.0
woopress/woocommerce/single-product/sale-flash.php
woopress/woocommerce/single-product/short-description.php
woopress/woocommerce/single-product/tabs/tabs.php
La  version 1.6.4 de woopress/woocommerce/single-product/up-sells.php está obsoleta. La versión del núcleo es la 3.0.0
La  version 2.3.2 de woopress/woocommerce/single-product-reviews.php está obsoleta. La versión del núcleo es la 3.2.0
woopress/woocommerce/single-product.php

Outdated Templates: ❌Aprende cómo actualizar
`

**Comment by claudiulodro** [comment]
I think this is caused by another plugin/theme you have active. 

On line 77 you see `global $product;`. Something is setting that to an array instead of a Product object, and this is why you are seeing that error. I'm unable to reproduce this problem on a site running only the WooCommerce plugin. Without the plugin list from your status report that's really all I can do. My advice is to try deactivating your other plugins (especially if you have any that work with or enhance reports) and see if it works with only WooCommerce active.

**Comment by foxvor** [comment]
PROBLEM SOLVED (hopefully):
First I debugged the $product array, and noticed that it had an "id" position, so first try was to change and use $product['id'] instead of $product->get_id(). This fixed the problem but only for the first column on the report. The second column replied with an error saying that couldn't treat $product object as an array. So now it was an object, not an array.

I noticed that this line was to set the $product object for each row or in case there isn't any yet. So I fixed it (hopefully not damaging anything else) by adding a condition to the if statement:

if ( ! $product || $product->get_id() !== $item->id ) {

changed to

if ( ! $product || is_array($product) || $product->get_id() !== $item->id ) {

this way, the first time this was called, $product would be an array and enter the if statement and be set as an object. This seems to be working so far. 

Hope this helps anyone else.
